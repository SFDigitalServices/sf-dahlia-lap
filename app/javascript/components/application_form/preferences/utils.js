
export const FIELD_NAME = 'shortFormPreferences'

export const buildFieldId = (i, field) => `${FIELD_NAME}.${i}.${field}`
