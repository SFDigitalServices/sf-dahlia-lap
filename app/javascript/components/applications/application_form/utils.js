export const mailingAddressFieldMap = {
  address: 'mailing_address',
  city: 'mailing_city',
  state: 'mailing_state',
  zip: 'mailing_zip'
}
