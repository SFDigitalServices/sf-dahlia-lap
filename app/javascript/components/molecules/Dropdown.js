import React from 'react'
import { find } from 'lodash'
import classNames from 'classnames'

import DropdownMenu from '../molecules/DropdownMenu'
import DropdownMenuMultiSelect from '../molecules/DropdownMenuMultiSelect'

class Dropdown extends React.Component {
  state = { expanded: false, style: { top: 40, left: 0 } }

  componentDidMount () {
    document.addEventListener('mousedown', this.handleClickOutside)
    document.addEventListener('keydown', this.onEscapeHandler)
  }

  componentWillUnmount () {
    document.removeEventListener('mousedown', this.handleClickOutside)
    document.removeEventListener('keydown', this.onEscapeHandler)
  }

  toggleExpand = (e) => {
    this.setState((prevState) => ({ expanded: !prevState.expanded }))
    // We need this to avoid triggering the collapse function from Foundation
    e.stopPropagation()
  }

  componentClickHandler = (e) => {
    // We want to hide the dropdown menu only when we click in the component but outside the menu
    if (this.wrapperRef === e.target) { this.setState({ expanded: false }) }
  }

  onChangeHandler = (value, label) => {
    if (!this.props.multiple) { this.setState({expanded: false}) }
    if (!this.props.multiple) { this.buttonRef.focus() }
    this.props.onChange && this.props.onChange(value, label)
  }

  menu = (classes) => {
    const { items, value, multiple } = this.props
    if (this.state.expanded) {
      if (multiple) {
        return (
          <DropdownMenuMultiSelect
            style={this.state.style}
            onChange={this.onChangeHandler}
            values={value}
            items={items}
            classes={classes} />
        )
      } else {
        return (
          <DropdownMenu
            style={this.state.style}
            onChange={this.onChangeHandler}
            value={value}
            items={items}
            classes={classes} />
        )
      }
    }
  }

  handleClickOutside = (event) => {
    if (this.wrapperRef && !this.wrapperRef.contains(event.target)) {
      this.setState({expanded: false})
    }
  }

 onEscapeHandler = (event) => {
   if (event.keyCode === 27) { this.setState({expanded: false}) }
 }

 render () {
   const {
     prompt,
     items,
     value,
     styles,
     buttonClasses,
     wrapperClasses,
     menuClasses = [],
     disabled
   } = this.props
   const selectedItem = find(items, { value: value })

   return (
     <div
       className={classNames('dropdown', wrapperClasses)}
       onClick={this.componentClickHandler}
       ref={(node) => { this.wrapperRef = node }}
       style={styles}>
       <button
         aria-expanded={this.state.expanded ? 'true' : 'false'}
         onClick={this.toggleExpand}
         ref={(node) => { this.buttonRef = node }}
         className={`button dropdown-button has-icon--right text-align-left ${buttonClasses ? buttonClasses.join(' ') : ''}`}
         type='button'
         disabled={disabled}>
         <span className='ui-icon ui-small'>
           <svg>
             <use xlinkHref='#i-arrow-down' />
           </svg>
         </span>
         {selectedItem ? selectedItem.label : prompt}
       </button>
       <div className='dropdown-menu-wrapper' aria-hidden={this.state.expanded ? 'false' : 'true'} role='menu'>
         {this.menu(menuClasses)}
       </div>
     </div>
   )
 }
}

export default Dropdown
