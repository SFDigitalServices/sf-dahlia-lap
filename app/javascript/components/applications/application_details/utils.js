import { each, startCase, isPlainObject, includes, last, replace } from 'lodash'
import utils from '~/utils/utils'

const cleanupWords = (value) => {
  each(
    [
      [' Or ', ' or '],
      [' Of ', ' of '],
      ['Proof', 'proof'],
      ['Who', 'who'],
      ['Ada', 'ADA'],
      [' To ', ' to '],
      ['Claimed', 'claimed']
    ],
    ([a, b]) => {
      value = replace(value, a, b)
    }
  )

  return value
}

export const getLabel = (field) => {
  let l = ''
  if (isPlainObject(field)) {
    l = utils.cleanField(field.label)
  } else {
    l = utils.cleanField(field)
  }
  if (includes(l, '.')) {
    l = last(l.split('.'))
  }
  l = utils.cleanField(l)

  return cleanupWords(startCase(l))
}

export const titleizeLabel = (label) => {
  return cleanupWords(startCase(label))
}
