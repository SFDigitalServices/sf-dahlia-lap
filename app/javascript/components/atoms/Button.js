import React from 'react'

import classNames from 'classnames'
import PropTypes from 'prop-types'

import { toPx } from 'utils/cssUtils'

const buttonSizeStyles = (clearVerticalPadding, minWidth) => {
  const widthStyle = minWidth
    ? {
        minWidth: toPx(minWidth)
      }
    : {}

  const heightStyle = clearVerticalPadding
    ? {
        paddingTop: 0,
        paddingBottom: 0
      }
    : {}

  return {
    ...heightStyle,
    ...widthStyle
  }
}

const buttonChildrenStyles = () => ({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-between',
  width: '100%'
})

const textStyles = (hasLeftIcon, hasRightIcon) => ({
  marginTop: '1rem',
  marginBottom: '1rem',
  marginRight: hasRightIcon && '1rem',
  marginLeft: hasLeftIcon && '1rem',
  flexGrow: 1
})

const iconWrapperStyles = () => ({
  flexShrink: 0,

  // we need to override font-size because of a css button selector that sets
  // the font size for all children. This is a problem because it adds 3px
  // to any icon wrapper.
  fontSize: 0
})

const wrapWithStyle = (node, style) => node && <span style={style}>{node}</span>

const Button = ({
  classes = null,
  iconLeft = null,
  iconRight = null,
  noBottomMargin,
  paddingHorizontal = 'normal',
  small = false,
  tertiary = false,
  text = null,
  tiny = false,
  type = 'button',
  minWidthPx = null,
  style = {},
  ...rest
}) => {
  const btnClassNames = classNames(classes, {
    'tight-padding': paddingHorizontal === 'tight',
    'extra-padding': paddingHorizontal === 'extra',
    small: small,
    tertiary: tertiary,
    tiny: tiny,
    'margin-bottom-none': noBottomMargin
  })

  const hasText = !!text

  return (
    <button
      className={btnClassNames}
      type={type}
      style={{ ...buttonSizeStyles(hasText, minWidthPx), ...style }}
      {...rest}
    >
      <div style={buttonChildrenStyles()}>
        {wrapWithStyle(iconLeft, iconWrapperStyles())}
        {wrapWithStyle(text, textStyles(!!iconLeft, !!iconRight))}
        {wrapWithStyle(iconRight, iconWrapperStyles())}
      </div>
    </button>
  )
}

Button.propTypes = {
  type: PropTypes.oneOf(['button', 'submit']),
  noBottomMargin: PropTypes.bool,
  classes: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)]),
  iconLeft: PropTypes.node,
  iconRight: PropTypes.node,
  minWidthPx: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  paddingHorizontal: PropTypes.oneOf(['normal', 'tight', 'extra']),
  small: PropTypes.bool,
  style: PropTypes.object,
  tertiary: PropTypes.bool,
  text: PropTypes.node,
  tiny: PropTypes.bool
}

export default Button
