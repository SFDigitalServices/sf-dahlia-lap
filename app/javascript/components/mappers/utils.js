import { omitBy, isUndefined } from 'lodash'
import moment from 'moment'

import { API_DATE_FORMAT } from 'utils/utils'

export const mapShape = (mapper, value) => {
  if (value) {
    return mapper(value)
  } else {
    return {}
  }
}

export const mapList = (mapper, list) => {
  if (list) {
    return list.map(mapper)
  } else {
    return []
  }
}

export const compactShape = (obj) => omitBy(obj, isUndefined)

export const domainDateOfBirthToApi = (dateOfBirth) => {
  // Convert domain (json) DOB to API format ('YYYY-MM-DD')
  // Create moment and reformat to ensure that the integers are padded.
  const DOB = `${dateOfBirth.year}-${dateOfBirth.month}-${dateOfBirth.day}`
  return dateOfBirth && moment(DOB, API_DATE_FORMAT).format(API_DATE_FORMAT)
}
