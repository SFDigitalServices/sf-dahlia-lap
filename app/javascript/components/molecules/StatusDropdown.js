import React from 'react'

import { LEASE_UP_STATUS_OPTIONS, getLeaseUpStatusClass } from '~/utils/statusUtils'
import Dropdown from '../molecules/Dropdown'

const StatusDropdown = ({ status, onChange, buttonClasses, menuClasses, wrapperClasses, styles, disabled }) => {
  let dropdownButtonClasses = [getLeaseUpStatusClass(status)]
  if (buttonClasses) dropdownButtonClasses.push(...buttonClasses)

  return (
    <Dropdown
      items={LEASE_UP_STATUS_OPTIONS}
      value={status}
      prompt='Status'
      onChange={onChange}
      buttonClasses={dropdownButtonClasses}
      menuClasses={menuClasses}
      wrapperClasses={wrapperClasses}
      styles={styles}
      disabled={disabled} />
  )
}

export default StatusDropdown
