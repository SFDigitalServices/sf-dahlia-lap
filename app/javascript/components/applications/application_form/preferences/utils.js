import { get, find, map, concat, pickBy } from 'lodash'

import { domainDateOfBirthToApi } from 'components/mappers/utils'

export const naturalKeyFromPreference = (p) => {
  return naturalKeyFromMember(get(p, 'application_member'))
}

export const naturalKeyFromMember = (member) => {
  return `${member.first_name},${member.last_name},${domainDateOfBirthToApi(member.date_of_birth)}`
}

export const memberNameFromPref = (id, householdMembers) => {
  const member = find(householdMembers, { id: id })
  if (member) {
    return `${member.first_name} ${member.last_name}`
  } else {
    return ''
  }
}

export const FIELD_NAME = 'preferences'

export const buildFieldId = (i, field) => `${FIELD_NAME}.${i}.${field}`

export const addNaturalKeyToPreference = (p) => {
  p.naturalKey = naturalKeyFromPreference(p)
}

export const buildHouseholdMembersOptions = (applicationMembers) => {
  return map(applicationMembers, (member) => {
    return {
      value: naturalKeyFromMember(member),
      label: `${member.first_name} ${member.last_name}`
    }
  })
}

export const getFullHousehold = (application) => {
  const fullHousehold = concat([application.applicant], application.household_members || [])
  return pickBy(
    fullHousehold,
    (m) =>
      // can only select someone for preference if they have name + DOB
      m && m.first_name && m.last_name && m.date_of_birth
  )
}
