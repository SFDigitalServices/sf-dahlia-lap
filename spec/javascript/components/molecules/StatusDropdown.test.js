import React from 'react'

import { mount, shallow } from 'enzyme'
import Select from 'react-select'

import Button from 'components/atoms/Button'
import Dropdown from 'components/molecules/Dropdown'
import StatusDropdown, { renderStatusOption } from 'components/molecules/StatusDropdown'
import { LEASE_UP_STATUS_OPTIONS } from 'utils/statusUtils'

const ON_CHANGE = jest.fn()

const getWrapper = (propOverrides = {}) =>
  mount(<StatusDropdown onChange={ON_CHANGE} {...propOverrides} />)

describe('StatusDropdown', () => {
  let wrapper
  test('it renders a select', () => {
    wrapper = getWrapper()
    expect(wrapper.find(Select).exists()).toBeTruthy()
  })

  test('it renders with default props correctly', () => {
    wrapper = getWrapper()

    const dropdownProps = wrapper.find(Dropdown).props()
    expect(dropdownProps.items).toEqual(LEASE_UP_STATUS_OPTIONS)
    expect(dropdownProps.value).toBeNull()
    expect(dropdownProps.placeholder).toEqual('Status')
    expect(dropdownProps.disabled).toBeFalsy()
  })

  describe('toggle', () => {
    test('renders as expected when value is null', () => {
      wrapper = getWrapper()
      const expectedButtonClasses = ['button', 'dropdown-button', 'tertiary']
      const toggleButton = wrapper.find(Select).find('.status-dropdown__control').find('button')
      expect(toggleButton.text()).toEqual('Status')
      expect(toggleButton.hasClass(expectedButtonClasses.join(' '))).toEqual(true)
      expect(toggleButton.prop('disabled')).toEqual(false)
    })

    test('renders without any additional button styles when none are provided', () => {
      wrapper = getWrapper()
      const toggleButton = wrapper.find(Select).find('.status-dropdown__control').find('button')
      expect(toggleButton.hasClass('expand')).toBe(false)
      expect(toggleButton.hasClass('tiny')).toBe(false)
      expect(toggleButton.hasClass('small')).toBe(false)
    })

    test('renders with left-aligned text', () => {
      wrapper = getWrapper({})

      expect(wrapper.find(Button).props().textAlign).toEqual('left')
    })

    test('renders with additional styles when provided', () => {
      wrapper = getWrapper({ expand: true, size: 'tiny' })
      let toggleButton = wrapper.find(Select).find('.status-dropdown__control').find('button')
      expect(toggleButton.hasClass('expand')).toBe(true)
      expect(toggleButton.hasClass('tiny')).toBe(true)
      expect(toggleButton.hasClass('small')).toBe(false)

      wrapper = getWrapper({ expand: true, size: 'small' })
      toggleButton = wrapper.find(Select).find('.status-dropdown__control').find('button')
      expect(toggleButton.hasClass('tiny')).toBe(false)
      expect(toggleButton.hasClass('small')).toBe(true)
    })

    test('renders value with styles when provided', () => {
      wrapper = getWrapper({ status: 'Appealed' })
      const toggleButton = wrapper.find(Select).find('.status-dropdown__control').find('button')
      expect(toggleButton.text()).toEqual('Appealed')
      expect(toggleButton.hasClass('is-appealed')).toBe(true)
    })
    test('does not render a value when forceDisplayPlaceholderText is true', () => {
      wrapper = getWrapper({
        placeholder: 'placeholder',
        status: 'Appealed',
        forceDisplayPlaceholderText: true
      })
      const toggleButton = wrapper.find(Select).find('.status-dropdown__control').find('button')
      expect(toggleButton.text()).toEqual('placeholder')
      expect(toggleButton.hasClass('is-appealed')).toBe(false)
    })

    test('renders when the dropdown is disabled', () => {
      wrapper = getWrapper({ disabled: true })
      const toggleButton = wrapper.find(Select).find('.status-dropdown__control').find('button')
      expect(toggleButton.prop('disabled')).toEqual(true)
    })

    test('renders a placeholder when provided', () => {
      wrapper = getWrapper({ placeholder: 'placeholder' })
      const toggleButton = wrapper.find(Select).find('.status-dropdown__control').find('button')
      expect(toggleButton.text()).toEqual('placeholder')
    })
  })

  describe('renderStatusOption', () => {
    const getStatusOptionWrapper = (option, selectedValue) =>
      shallow(renderStatusOption(option, { selectValue: [{ value: selectedValue }] }))
    const testValue = 'testValue'
    const testStatusClassName = 'status-class-name'
    const testLabel = 'testLabel'
    const sampleOption = {
      value: testValue,
      label: testLabel,
      statusClassName: testStatusClassName
    }

    test('renders a styled value with expected styles', () => {
      wrapper = getStatusOptionWrapper(sampleOption, 'notSelected')
      expect(wrapper.find('li').hasClass(testStatusClassName)).toEqual(true)
      expect(wrapper.find('li').hasClass('dropdown-menu_item')).toEqual(true)
      expect(wrapper.text()).toEqual(testLabel)
      // Verify that its not selected
      expect(wrapper.find('li').prop('aria-selected')).toEqual(false)
    })

    test('renders a selected value as expected', () => {
      wrapper = getStatusOptionWrapper(sampleOption, testValue)
      expect(wrapper.find('li').prop('aria-selected')).toEqual(true)
      expect(wrapper.text()).toEqual(testLabel)
    })

    test('renders as expected with a null value', () => {
      wrapper = getStatusOptionWrapper({}, testValue)
      expect(wrapper.find('li').hasClass(testStatusClassName)).toEqual(false)
      expect(wrapper.text()).toEqual('')
    })
  })
})
