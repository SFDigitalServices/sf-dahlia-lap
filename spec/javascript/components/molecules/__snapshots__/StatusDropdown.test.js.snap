// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusDropdown toggle renders as expected when value is null 1`] = `
<DocumentFragment>
  <div
    class="dropdown css-b62m3t-container"
  >
    <span
      class="css-1f43avz-a11yText-A11yText"
      id="react-select-4-live-region"
    />
    <span
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="css-1f43avz-a11yText-A11yText"
      role="log"
    />
    <div
      class="status-dropdown__control css-uuqjpg-Control"
    >
      <button
        class="button dropdown-button tertiary margin-bottom-none"
        style="padding-top: 0px; padding-bottom: 0px;"
        type="button"
      >
        <div
          style="display: flex; align-items: center; justify-content: space-between; width: 100%;"
        >
          <span
            style="margin-top: 1rem; margin-bottom: 1rem; margin-right: 1rem; overflow: auto; flex-grow: 1; text-align: left;"
          >
            Status
          </span>
          <div
            style="width: 0px; height: 0px;"
          >
            <div
              class="status-dropdown__value-container css-1fdsijx-ValueContainer"
            >
              <div
                class="status-dropdown__placeholder css-1jqq78o-placeholder"
                id="react-select-4-placeholder"
              />
              <input
                aria-activedescendant=""
                aria-autocomplete="list"
                aria-describedby="react-select-4-placeholder"
                aria-expanded="false"
                aria-haspopup="true"
                aria-readonly="true"
                class="css-mohuvp-dummyInput-DummyInput"
                id="react-select-4-input"
                inputmode="none"
                role="combobox"
                tabindex="0"
                value=""
              />
            </div>
          </div>
          <span
            style="flex-shrink: 0; font-size: 0px;"
          >
            <svg
              style="width: .75rem; height: .75rem;"
            >
              <use
                xlink:href="#i-arrow-down"
              />
            </svg>
          </span>
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`StatusDropdown toggle renders with additional styles when provided 1`] = `
<DocumentFragment>
  <div
    class="dropdown css-b62m3t-container"
  >
    <span
      class="css-1f43avz-a11yText-A11yText"
      id="react-select-7-live-region"
    />
    <span
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="css-1f43avz-a11yText-A11yText"
      role="log"
    />
    <div
      class="status-dropdown__control css-uuqjpg-Control"
    >
      <button
        class="button dropdown-button expand tertiary tight-padding tiny margin-bottom-none"
        style="padding-top: 0px; padding-bottom: 0px;"
        type="button"
      >
        <div
          style="display: flex; align-items: center; justify-content: space-between; width: 100%;"
        >
          <span
            style="margin-top: 0.5rem; margin-bottom: 0.5rem; margin-right: 0.5rem; overflow: auto; flex-grow: 1; text-align: left;"
          >
            Status
          </span>
          <div
            style="width: 0px; height: 0px;"
          >
            <div
              class="status-dropdown__value-container css-1fdsijx-ValueContainer"
            >
              <div
                class="status-dropdown__placeholder css-1jqq78o-placeholder"
                id="react-select-7-placeholder"
              />
              <input
                aria-activedescendant=""
                aria-autocomplete="list"
                aria-describedby="react-select-7-placeholder"
                aria-expanded="false"
                aria-haspopup="true"
                aria-readonly="true"
                class="css-mohuvp-dummyInput-DummyInput"
                id="react-select-7-input"
                inputmode="none"
                role="combobox"
                tabindex="0"
                value=""
              />
            </div>
          </div>
          <span
            style="flex-shrink: 0; font-size: 0px;"
          >
            <svg
              style="width: .75rem; height: .75rem;"
            >
              <use
                xlink:href="#i-arrow-down"
              />
            </svg>
          </span>
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`StatusDropdown toggle renders with additional styles when provided 2`] = `
<DocumentFragment>
  <div
    class="dropdown css-b62m3t-container"
  >
    <span
      class="css-1f43avz-a11yText-A11yText"
      id="react-select-7-live-region"
    />
    <span
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="css-1f43avz-a11yText-A11yText"
      role="log"
    />
    <div
      class="status-dropdown__control css-uuqjpg-Control"
    >
      <button
        class="button dropdown-button expand tertiary small margin-bottom-none"
        style="padding-top: 0px; padding-bottom: 0px;"
        type="button"
      >
        <div
          style="display: flex; align-items: center; justify-content: space-between; width: 100%;"
        >
          <span
            style="margin-top: 1rem; margin-bottom: 1rem; margin-right: 1rem; overflow: auto; flex-grow: 1; text-align: left;"
          >
            Status
          </span>
          <div
            style="width: 0px; height: 0px;"
          >
            <div
              class="status-dropdown__value-container css-1fdsijx-ValueContainer"
            >
              <div
                class="status-dropdown__placeholder css-1jqq78o-placeholder"
                id="react-select-7-placeholder"
              />
              <input
                aria-activedescendant=""
                aria-autocomplete="list"
                aria-describedby="react-select-7-placeholder"
                aria-expanded="false"
                aria-haspopup="true"
                aria-readonly="true"
                class="css-mohuvp-dummyInput-DummyInput"
                id="react-select-7-input"
                inputmode="none"
                role="combobox"
                tabindex="0"
                value=""
              />
            </div>
          </div>
          <span
            style="flex-shrink: 0; font-size: 0px;"
          >
            <svg
              style="width: .75rem; height: .75rem;"
            >
              <use
                xlink:href="#i-arrow-down"
              />
            </svg>
          </span>
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`StatusDropdown toggle renders with left-aligned text 1`] = `
<DocumentFragment>
  <div
    class="dropdown css-b62m3t-container"
  >
    <span
      class="css-1f43avz-a11yText-A11yText"
      id="react-select-6-live-region"
    />
    <span
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="css-1f43avz-a11yText-A11yText"
      role="log"
    />
    <div
      class="status-dropdown__control css-uuqjpg-Control"
    >
      <button
        class="button dropdown-button tertiary margin-bottom-none"
        style="padding-top: 0px; padding-bottom: 0px;"
        type="button"
      >
        <div
          style="display: flex; align-items: center; justify-content: space-between; width: 100%;"
        >
          <span
            style="margin-top: 1rem; margin-bottom: 1rem; margin-right: 1rem; overflow: auto; flex-grow: 1; text-align: left;"
          >
            Status
          </span>
          <div
            style="width: 0px; height: 0px;"
          >
            <div
              class="status-dropdown__value-container css-1fdsijx-ValueContainer"
            >
              <div
                class="status-dropdown__placeholder css-1jqq78o-placeholder"
                id="react-select-6-placeholder"
              />
              <input
                aria-activedescendant=""
                aria-autocomplete="list"
                aria-describedby="react-select-6-placeholder"
                aria-expanded="false"
                aria-haspopup="true"
                aria-readonly="true"
                class="css-mohuvp-dummyInput-DummyInput"
                id="react-select-6-input"
                inputmode="none"
                role="combobox"
                tabindex="0"
                value=""
              />
            </div>
          </div>
          <span
            style="flex-shrink: 0; font-size: 0px;"
          >
            <svg
              style="width: .75rem; height: .75rem;"
            >
              <use
                xlink:href="#i-arrow-down"
              />
            </svg>
          </span>
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`StatusDropdown toggle renders without any additional button styles when none are provided 1`] = `
<DocumentFragment>
  <div
    class="dropdown css-b62m3t-container"
  >
    <span
      class="css-1f43avz-a11yText-A11yText"
      id="react-select-5-live-region"
    />
    <span
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="css-1f43avz-a11yText-A11yText"
      role="log"
    />
    <div
      class="status-dropdown__control css-uuqjpg-Control"
    >
      <button
        class="button dropdown-button tertiary margin-bottom-none"
        style="padding-top: 0px; padding-bottom: 0px;"
        type="button"
      >
        <div
          style="display: flex; align-items: center; justify-content: space-between; width: 100%;"
        >
          <span
            style="margin-top: 1rem; margin-bottom: 1rem; margin-right: 1rem; overflow: auto; flex-grow: 1; text-align: left;"
          >
            Status
          </span>
          <div
            style="width: 0px; height: 0px;"
          >
            <div
              class="status-dropdown__value-container css-1fdsijx-ValueContainer"
            >
              <div
                class="status-dropdown__placeholder css-1jqq78o-placeholder"
                id="react-select-5-placeholder"
              />
              <input
                aria-activedescendant=""
                aria-autocomplete="list"
                aria-describedby="react-select-5-placeholder"
                aria-expanded="false"
                aria-haspopup="true"
                aria-readonly="true"
                class="css-mohuvp-dummyInput-DummyInput"
                id="react-select-5-input"
                inputmode="none"
                role="combobox"
                tabindex="0"
                value=""
              />
            </div>
          </div>
          <span
            style="flex-shrink: 0; font-size: 0px;"
          >
            <svg
              style="width: .75rem; height: .75rem;"
            >
              <use
                xlink:href="#i-arrow-down"
              />
            </svg>
          </span>
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
